src/api/v1/statuses.test.ts

```diff
@@ -0,0 +1,172 @@
+import { beforeEach, describe, expect, it } from "vitest";
+
+import { cleanDatabase } from "../../../tests/helpers";
+import {
+  bearerAuthorization,
+  createAccount,
+  createOAuthApplication,
+  getAccessToken,
+  getApplication,
+} from "../../../tests/helpers/oauth";
+
+import app from "../../index";
+
+describe.sequential("/api/v1/accounts/verify_credentials", () => {
+  let client: Awaited<ReturnType<typeof createOAuthApplication>>;
+  let account: Awaited<ReturnType<typeof createAccount>>;
+  let application: Awaited<ReturnType<typeof getApplication>>;
+  let accessToken: Awaited<ReturnType<typeof getAccessToken>>;
+
+  beforeEach(async () => {
+    await cleanDatabase();
+
+    account = await createAccount({ generateKeyPair: true });
+    client = await createOAuthApplication({
+      scopes: ["write"],
+    });
+    application = await getApplication(client);
+    accessToken = await getAccessToken(client, account, ["write"]);
+  });
+
+  it("Successfully creates a new status with a valid access token using JSON", async () => {
+    expect.assertions(7);
+
+    const body = JSON.stringify({
+      status: "Hello world",
+      media_ids: [],
+    });
+
+    const response = await app.request("/api/v1/statuses", {
+      method: "POST",
+      headers: {
+        authorization: bearerAuthorization(accessToken),
+        "Content-Type": "application/json",
+      },
+      body: body,
+    });
+
+    expect(response.status).toBe(200);
+    expect(response.headers.get("content-type")).toBe("application/json");
+    expect(response.headers.get("access-control-allow-origin")).toBe("*");
+
+    const json = await response.json();
+
+    expect(typeof json).toBe("object");
+    expect(json.content).toBe("<p>Hello world</p>\n");
+    expect(json.account.id).toBe(account.id);
+    expect(json.application.name).toBe(application.name);
+  });
+
+  it("Successfully creates a new status with a valid access token using FormData", async () => {
+    expect.assertions(7);
+
+    const body = new FormData();
+    body.append("status", "Hello world");
+
+    const response = await app.request("/api/v1/statuses", {
+      method: "POST",
+      headers: {
+        authorization: bearerAuthorization(accessToken),
+      },
+      body: body,
+    });
+
+    expect(response.status).toBe(200);
+    expect(response.headers.get("content-type")).toBe("application/json");
+    expect(response.headers.get("access-control-allow-origin")).toBe("*");
+
+    const json = await response.json();
+
+    expect(typeof json).toBe("object");
+    expect(json.content).toBe("<p>Hello world</p>\n");
+    expect(json.account.id).toBe(account.id);
+    expect(json.application.name).toBe(application.name);
+  });
+
+  it("Can update a status using JSON", async () => {
```

Comment:
These tests are a little messy, but that's due to us not having a good way to mock a status object in the database, so instead I'm just creating first, then updating.